core/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -509,6 +509,22 @@ public Object readGenericValue() throws IOException {
         return namedWriteable.readFrom(this);
     }
 
+    /**
+     * Reads a list of {@link NamedWriteable} from the current stream, by first reading its size and then
+     * reading the individual objects using {@link #readNamedWriteable()}.
+     */
+    public <C> List<C>  readNamedWritableList() throws IOException {
+        List<C> list = new ArrayList<>();
+        int size = readInt();
+        for (int i = 0; i < size; i++) {
+            C obj = readNamedWriteable();
+            if (obj != null) {
```

Comment:
can it ever be null? I think there is something wrong if it is null, cause it should always throw exception upfront in case of issues?